index.js

```diff
@@ -46,230 +48,230 @@ function resolveInputDirectory(inputNodes) {
- delete results.results;
- results.log = results.output;
- Object.assign(results, resultsInner);
- results.source = relativePath;
- results.output = '';
- return results;
-};
 
-/**
-  * @method testGenerator
-  *
-  *  Alias of escapeString for hooks
-  */
-StyleLinter.prototype.escapeErrorString = escapeString;
+  /**
+    * @method postProcess
+    * This method is called after, the file has been linted:
+    *  - Logs to console
+    *  - Generate tests
+    * @override
+    */
+  postProcess(results, relativePath) {
+    if(results) {
+      if(results.errored){
+        if(this.onError) {
+        this.onError(results);
+        }
+        if(this.log)
+        this.consoleLogger(results, relativePath);
+      }
+      return results;
+    } else {
+      return {};
+    }
+  }
 
+  /**
+    * @method processResults
+    *
+    *  Reformats default results object
+    *  {
+    *   errored: boolean if file errored or not,
+    *   output: String contains test if generate test is true,
+    *   log: String default logging string,
+    *   source: String relitivePath,
+    *   deprecations: Array of errors,
+    *   invalidOptionWarnings: Array,
+    *   warnings: Array of errors,
+    *   ignored: Array ignored files,
+    *   _postcssResult: Object for postcss
+    *  }
+    */
+  processResults(results, relativePath) {
+  let resultsInner = results.results[0];
+  resultsInner.errored = results.errored;
+  resultsInner.source = relativePath;
+  delete results.results;
+  results.log = results.output;
+  Object.assign(results, resultsInner);
+  results.source = relativePath;
+  results.output = '';
+  return results;
+  };
 
+  /**
+    * @method testGenerator
+    *
+    *  Alias of escapeString for hooks
+    */
+  escapeErrorString() {
+    return escapeString.apply(this, arguments);
+  }
 
-/**
-  * @method consoleLogger
-  *
-  *  custom console logger
-  */
-StyleLinter.prototype.consoleLogger = function(results, relativePath) {
-  this.console.log(results.log);
-};
+  /**
+    * @method consoleLogger
+    *
+    *  custom console logger
+    */
+  consoleLogger(results, relativePath) {
+    this.console.log(results.log);
+  }
 
-/**
-  * @method testGenerator
-  *
-  *  Geneartes tests.
-  */
-StyleLinter.prototype.testGenerator = function(relativePath, errors) {
-  var assertions = [];
-  var module  = "module('Style Lint');\n";
-  var test = "test('" + relativePath + " should pass stylelint', function() {\n";
-  if(!errors){
-    var assertion =  "  ok(\'true , "+relativePath+" passed stylelint\');";
-    return module+test+assertion+"\n});\n";
-  } else {
-    for(var i = 0; i < errors.warnings.length; i++){
-      var warning = errors.warnings[i];
-      var index = warning.line+':'+warning.column;
-      assertions.push("  ok(" + false + ", '"+index +" "+this.escapeErrorString(warning.text)+"');");
+  /**
+    * @method testGenerator
+    *
+    *  Geneartes tests.
+    */
+  testGenerator(relativePath, errors) {
```

Comment:
can make the file less intimidating if we move this function to a separate file.

Reply:
I would ultimately like to drop it completely once we have implemented https://github.com/billybonks/broccoli-stylelint/issues/40

Reply:
👍 hopefully can drop some of this at the same time. https://github.com/billybonks/ember-cli-stylelint/blob/master/index.js#L47

Reply:
i noticed that ember-eslint doesn't use `project.generateTestFile`

Reply:
yep, exactly. all of that can be dropped then.

Reply:
I wanted to get the code in a more readable state first though before attempting that refactoring